state/machine_linklayerdevices.go

```diff
@@ -206,18 +206,27 @@ func (m *Machine) SetLinkLayerDevices(devicesArgs ...LinkLayerDeviceArgs) (err e
 			}
 		}
 
+		// We've checked the model is alive directly, and we assert the machine is alive, we don't need to also
+		// assert the model is alive, because then the machine would be dying as well.
 		ops := []txn.Op{
-			assertModelActiveOp(m.st.ModelUUID()),
 			m.assertAliveOp(),
 		}
 
 		setDevicesOps, err := m.setDevicesFromDocsOps(newDocs)
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
+		if len(setDevicesOps) == 0 {
+			// No need to assert only that the machine is alive
+			logger.Debugf("no changes to LinkLayerDevices for machine %q", m.Id())
+			return nil, errNoChanges
```

Comment:
thanks, I expected something like that, but there wasn't anything like that around this code.